kubernetes/nodecollection.go

```diff
@@ -153,7 +172,22 @@ func ensureNodeSource(config KubeAgentConfig) (KubeAgentConfig, error) {
 		return config, fmt.Errorf("error retrieving nodes: %s", err)
 	}
 
-	ip, port, err := clientSetNodeSource.NodeAddress(&nodes.Items[0])
+	// Some nodes may not be ready, we want to find the first node that is ready so we can set the node source
+	var firstReadyNode *v1.Node
+
+	for _, n := range nodes.Items {
+		nodeReady := v1Node.IsNodeReady(&n)
+
+		if nodeReady {
+			firstReadyNode = &n
+		}
+	}
```

Comment:
It might be more efficient to break out of the loop on the first ready node found, rather than processing the whole list. Pulling that into a function that just returns a firstReadyNode would also be a drop-in way to lower the cyclomatic complexity.....the most annoying lint 😆

Reply:
oh yes it is the most annoying lint, feel like sometimes if you fix it you just potentially make the code less readable. I will take a look at this!